diff --git a/node_modules/react-native-google-places-autocomplete/GooglePlacesAutocomplete.js b/node_modules/react-native-google-places-autocomplete/GooglePlacesAutocomplete.js
index 60836de..5aa7017 100644
--- a/node_modules/react-native-google-places-autocomplete/GooglePlacesAutocomplete.js
+++ b/node_modules/react-native-google-places-autocomplete/GooglePlacesAutocomplete.js
@@ -2,7 +2,7 @@
 import debounce from 'lodash.debounce';
 import PropTypes from 'prop-types';
 import Qs from 'qs';
-import { v4 as uuidv4 } from 'uuid';
+// import { v4 as uuidv4 } from 'uuid';
 import React, {
   forwardRef,
   useMemo,
@@ -25,6 +25,7 @@ import {
   TextInput,
   View,
 } from 'react-native';
+import * as Crypto from 'expo-crypto';
 
 const defaultStyles = {
   container: {
@@ -85,6 +86,21 @@ export const GooglePlacesAutocomplete = forwardRef((props, ref) => {
       return false;
     }
   };
+  async function generateUUID() {
+    const randomBytes = await Crypto.getRandomBytesAsync(16);
+    randomBytes[6] = (randomBytes[6] & 0x0f) | 0x40; // Version 4
+    randomBytes[8] = (randomBytes[8] & 0x3f) | 0x80; // Variant 10
+  
+    const byteToHex = Array.from(randomBytes).map(b => ('00' + b.toString(16)).slice(-2));
+  
+    return [
+      byteToHex.slice(0, 4).join(''),
+      byteToHex.slice(4, 6).join(''),
+      byteToHex.slice(6, 8).join(''),
+      byteToHex.slice(8, 10).join(''),
+      byteToHex.slice(10, 16).join(''),
+    ].join('-');
+  }
 
   const buildRowsFromResults = useCallback(
     (results, text) => {
@@ -160,7 +176,7 @@ export const GooglePlacesAutocomplete = forwardRef((props, ref) => {
   const [listLoaderDisplayed, setListLoaderDisplayed] = useState(false);
 
   const inputRef = useRef();
-  const [sessionToken, setSessionToken] = useState(uuidv4());
+  const [sessionToken, setSessionToken] = useState(generateUUID());
   useEffect(() => {
     setUrl(getRequestUrl(props.requestUrl));
   }, [getRequestUrl, props.requestUrl]);
@@ -337,7 +353,7 @@ export const GooglePlacesAutocomplete = forwardRef((props, ref) => {
               fields: props.fields,
             }),
         );
-        setSessionToken(uuidv4());
+        setSessionToken(generateUUID());
       } else {
         request.open(
           'GET',
